### PR TITLE
Expand upon DOM section in PHP 8.4 new features

### PR DESCRIPTION
**Pull request by nielsdos**
cc @TimWolla

### CODE REVIEW

**Review by TimWolla — COMMENTED** [review]
Text makes sense to me. Should also be added to UPGRADING then.

Leaving the final review to someone better with English and Docbook.

File: appendices/migration84/new-features.xml [hunk 1]
```diff
@@ -277,7 +288,10 @@ $object = $reflector->newLazyGhost($initializer);
   <simpara>
    It is now possible to pass any callable to
    <methodname>DOMXPath::registerPhpFunctions</methodname>.
-   <!-- TODO Mention DOMXPath::registerPHPFunctionNS ? -->
+
+   Additionally, it is now possible to register callbacks using native
+   syntax instead of passing the method name as a string; by using
+   <methodname>DOMXPath::registerPhpFunctionNs</methodname>.
```
Comment (cmb69) [inline appendices/migration84/new-features.xml:294]: That semicolon looks strange to me. Maybe reword:
```suggestion
   Additionally, by using <methodname>DOMXPath::registerPhpFunctionNs</methodname>,
   it is now possible to register callbacks using native
   syntax instead of passing the method name as a string.
```
Still, feels not quite right.

**Review by cmb69 — COMMENTED** [review]
Thank you!

File: appendices/migration84/new-features.xml [hunk 2]
```diff
@@ -260,6 +260,17 @@ $object = $reflector->newLazyGhost($initializer);
  <sect2 xml:id="migration84.new-features.dom">
   <title>DOM</title>
 
+  <!-- RFC: https://wiki.php.net/rfc/domdocument_html5_parser -->
+  <!-- RFC: https://wiki.php.net/rfc/opt_in_dom_spec_compliance -->
+  <para>
+   Added the Dom namespace with new classes as counterparts to the existing
```
[the remark this answers is not in the document]
Comment (Girgias) [reply appendices/migration84/new-features.xml:266]: Could maybe use the `<package>` DocBook tag, but I don't know if we do any processing on it.

[the remark this answers is not in the document]
Comment (jimwins) [reply appendices/migration84/new-features.xml:266]: Looks like we don't use `<package>` at all yet. PhD will just make it `<strong>`, but that's easy to change.

[the remark this answers is not in the document]
Comment (nielsdos) [reply appendices/migration84/new-features.xml:266]: I think package is most suiting, or at least more than literal or code.

File: appendices/migration84/new-features.xml (hunk 1 above)
Comment (nielsdos) [reply appendices/migration84/new-features.xml]: I tried rewording a bit.

File: appendices/migration84/new-features.xml [hunk 3]
```diff
@@ -277,7 +288,10 @@ $object = $reflector->newLazyGhost($initializer);
   <simpara>
    It is now possible to pass any callable to
    <methodname>DOMXPath::registerPhpFunctions</methodname>.
-   <!-- TODO Mention DOMXPath::registerPHPFunctionNS ? -->
+
+   Furthermore, with <methodname>DOMXPath::registerPhpFunctionNs</methodname>,
+   you can now register callbacks that will use native function call syntax
```
Comment (cmb69) [inline appendices/migration84/new-features.xml:293]: No personalization please:

```suggestion
   callbacks can now be registered that will use native function call syntax
```

**Review by cmb69 — APPROVED** [review]

File: appendices/migration84/new-features.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Girgias) [reply appendices/migration84/new-features.xml]: Could always add a `role="namespace"` attribute.

Which for PIE/PECL would be `role="PIE"`/`role="PECL"`.

File: appendices/migration84/new-features.xml [hunk 4]
```diff
@@ -260,6 +260,17 @@ $object = $reflector->newLazyGhost($initializer);
  <sect2 xml:id="migration84.new-features.dom">
   <title>DOM</title>
 
+  <!-- RFC: https://wiki.php.net/rfc/domdocument_html5_parser -->
+  <!-- RFC: https://wiki.php.net/rfc/opt_in_dom_spec_compliance -->
+  <para>
```
Comment (Girgias) [inline appendices/migration84/new-features.xml:265]:
```suggestion
  <simpara>
```

**Review by Girgias — APPROVED** [review]
Minor nit, but LGTM
